include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -5706,14 +5719,171 @@ FEEvaluationAccess<dim, dim, Number, is_face, VectorizedArrayType>::operator=(
   return *this;
 }
 
+namespace internal
+{
+  template <int dim, bool is_face>
+  inline const unsigned int *
+  get_index(const std::uint8_t fn)
+  {
+    static const unsigned int index0[3] = {0, 1, 2};
+    static const unsigned int index1[3] = {1, 0, 2};
+    static const unsigned int index2[3] = {2, 0, 1};
+    static const unsigned int index3[3] = {1, 2, 0};
```

Comment:
I didn't really want to call this function more than once, and I figured that I only wanted to return a 1D array. But maybe I should do like what is done in `reorder_face_derivative_indices`, and call it for every index?

Reply:
I think that a static `std::array` should not be all that different from the pointer to a 1D array, as long as the compiler inlines everything correctly. I can't say whether it makes sense to call a function similar to `reorder_face_derivative_indices` for every index without benchmarking or looking at the assembly output of a compiler, as the function here is different from the one in `MappingInfo` in the sense that this one here is a hot loop, whereas the other function is "only" a setup function and thus potentially not used as often. That was also the intent for the original design by @kkormann and me, we tried to move everything expensive like shuffling around indices to the setup function to have as simple data structures in the actual operator evaluation for the regular DG face integrals.

Obviously, here we need to apply the reordering for every point in the evaluation. Or do we? Right now it is only curiosity, but do you think one could simplify the evaluators by shuffling around the normal and tangential components of the solution in such a way that the re-ordering of the Jacobian matrix on the faces would be the correct one again? I could even imagine that this might somewhat reduce the number of cases in the `evaluation_kernels` for the faces again, at least in terms of polynomial degrees that need to be expanded in normal and tangential direction. What do you think?

Reply:
I do think you are correct! I see no immediate reason why we wouldn't be able to shuffle the components like we shuffle the derivatives in `gradients_quad` for instance, which is why we don't have to shuffle the inverse jacobian in the get/submit functions. I'll look into it.

Reply:
Just fyi, it really made the function in `evaluation_kernels` muuuch nicer! Great idea @kronbichler, thanks! :)

Now I also (finally) see the value of having zx-coordinates  for the faces